imap_processing/swe/tests/test_swe_science.py

```diff
@@ -1,15 +1,67 @@
+import pandas as pd
 import pytest
 
+from imap_processing import imap_module_directory
 from imap_processing.swe.l0 import decom_swe
 from imap_processing.swe.l1a.swe_science import swe_science, uncompress_counts
 
 
 @pytest.fixture(scope="session")
 def decom_test_data():
     """Read test data from file"""
-    packet_file = "imap_processing/swe/tests/science_block_20221116_163611Z_idle.bin"
-    data_packet_list = decom_swe.decom_packets(packet_file)
-    return data_packet_list
+    # NOTE: data was provided in this sequence in both bin and validation data
+    # from instrument team.
+    # Packet 1 has spin 4's data
+    # Packet 2 has spin 1's data
+    # Packet 3 has spin 2's data
+    # Packet 4 has spin 3's data
+    packet_files = [
+        f"{imap_module_directory}/swe/tests/science_packets/20230927173238_SWE_SCIENCE_packet.bin",
+        f"{imap_module_directory}/swe/tests/science_packets/20230927173253_SWE_SCIENCE_packet.bin",
+        f"{imap_module_directory}/swe/tests/science_packets/20230927173308_SWE_SCIENCE_packet.bin",
+        f"{imap_module_directory}/swe/tests/science_packets/20230927173323_SWE_SCIENCE_packet.bin",
+    ]
+    data = []
+    for packet_file in packet_files:
+        data.extend(decom_swe.decom_packets(packet_file))
+    return data
+
+
+def test_total_packets_in_data_file(decom_test_data):
+    """Test if total packets in data file is correct"""
+    total_packets = 4
+    assert len(decom_test_data) == total_packets
+
+
+def test_ccsds_header(decom_test_data):
```

Comment:
does `space_packet_parser` do this for us automatically? I thought it complained right away if the header was not valid.

Reply:
`space_packet_parser` does give an error but not very good error message.